ha/k8s_setup/ha_setup.py

```diff
@@ -0,0 +1,342 @@
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+        args = parser.parse_args(argv)
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """
+        Add Command args for parsing.
+        """
+        setup_arg_parser = parser.add_parser(cls.name, help='setup %s' % name)
+        setup_arg_parser.add_argument('--config', help='Config URL', required=True)
+        setup_arg_parser.add_argument('--services', help='service name', required=True)
+        setup_arg_parser.add_argument('args', nargs='*', default=[], help='args')
+        setup_arg_parser.set_defaults(command=cls)
+
+    @staticmethod
+    def remove_file(file: str):
+        """
+        Check if file exist and delete existing file.
+
+        Args:
+            file ([str]): File or Dir name to be deleted.
+        """
+        if os.path.exists(file):
+            if os.path.isfile(file):
+                os.remove(file)
+            elif os.path.isdir(file):
+                shutil.rmtree(file)
+            else:
+                raise SetupError(f"{file} is not dir and file, can not be deleted.")
+
+class PostInstallCmd(Cmd):
+    """
+    PostInstall Setup Cmd
+    """
+    name = "post_install"
+
+    def __init__(self, args: dict):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process post_install command.
+        """
+        sys.stdout.write("All post install checks for HA are upto date.\n")
+
+class PrepareCmd(Cmd):
+    """
+    Prepare Setup Cmd
+    """
+    name = "prepare"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process prepare command.
+        """
+        sys.stdout.write("HA is prepared for performing mini provisioning.\n")
+
+class ConfigCmd(Cmd):
+    """
+    Setup Config Cmd
+    """
+    name = "config"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process config command.
+        """
+        try:
+            consul_endpoint = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints[0]')
+            if not consul_endpoint:
+                sys.stderr.write(f'Failed to get consul config. consul_config: {consul_endpoint}. \n')
+                sys.exit(1)
+            # TODO: Uncomment whenever prometheus config will be available
+            # Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}prometheus{_DELIM}endpoints[0]')
+
+            # command can be used to get scrape interval of prometheus
+            # kubectl describe configmap -n cortx | grep scrape_interval | awk \'{print $2}\'
+
+            conf_file_dict = {'LOG' : {'path' : const.HA_LOG_DIR, 'level' : const.HA_LOG_LEVEL},
```

Comment:
can you add a TODO here if LOG_LEVLE is to be made configurable

Reply:
I think, in Log.init, you can give whatever level you want. Why to want seperate TODO here?